activestorage/test/service/configurations.example.yml

```diff
@@ -27,3 +27,11 @@
 #   storage_account_name: ""
 #   storage_access_key: ""
 #   container: ""
+#
+# azure_aad:
+#   service: AzureStorage
+#   storage_account_name: ""
+#   container: ""
+#   tenant_id: aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa
+#   client_id: bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb
+#   client_secret: cccccccc-cccc-cccc-cccc-cccccccccccc
```

Comment:
Do you already have a storage account you're using on CI? If so, all we need to do is create a service principal and grant it the appropriate IAM permissions to the storage account.

You can do this via a snippet similar to the one I posted in the PR description :

```sh
service_principal_name="enter some name here"
storage_account_name="enter your storage account name here"
container_name="enter the name for the container used for the new tests here"

# fetch the resource id of the existing storage account
storage_account_id="$(az storage account show --name "${storage_account_name}" --query id --output tsv)"

# create the container that the new tests will use
# we need to do this as a one-time setup step since for extra security we'll only grant
# our service principal access to this specific container instead of the entire storage account
# which means that the service principal won't be able to dynamically create new containers
az storage container create --name "${container_name}" --account-name "${storage_account_name}"

# create the service principal we'll be using for the tests
sp="$(az ad sp create-for-rbac --name "${service_principal_name}" --skip-assignment --output json)"
sp_tenant="$(jq -r '.tenant' <<< "${sp}")"
sp_client="$(jq -r '.appId' <<< "${sp}")"
sp_secret="$(jq -r '.password' <<< "${sp}")"

# grant the service principal the appropriate IAM permissions on the storage account
# the data contributor role is required for read/write/list/delete operations on blobs
# in the container and the delegator permission is required to create signed private urls
az role assignment create --role "Storage Blob Data Contributor" --assignee "${sp_client}" --scope "${storage_account_id}/blobServices/default/containers/${container_name}"
az role assignment create --role "Storage Blob Delegator" --assignee "${sp_client}" --scope "${storage_account_id}"
```

Alternatively you can also do this in the Azure Portal: see the [docs for creating a service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#create-an-azure-active-directory-application) and the [docs for assigning a storage role](https://docs.microsoft.com/en-us/azure/storage/common/storage-auth-aad-rbac-portal#assign-a-built-in-rbac-role).

If you do not already have a storage account that's being used in the CI or wish to create a new one, you can use the snippet that I posted in the pull request description to create all the required resources and update the `configurations.yml` file as required.

Hope this helps. Let me know what additional assistance I can provide.